_pages/dev/ssr/server-side-rendering-contextual-logging.md

```diff
@@ -0,0 +1,261 @@
+
+You can also customize the behavior of the `LoggerService` by providing your own implementation of the `LoggerService` interface.
+
+### LoggerService Behavior in an SSR Context
+
+For server-side rendering, Spartacus provides the `ExpressLoggerService` to the Angular context. This service extends the `LoggerService` and provides the `DefaultExpressServerLogger` for logging. Using this service guarantees that logs are properly formatted and contain the necessary context about the source of the log.
+
+**Note:** By default, Spartacus does not output any sensitive data in the logs. However, if you include sensitive data in your log messages, you should be aware that it will be visible in the logs. Accordingly, it is your responsibility to ensure that no sensitive data is included in the log messages.
+
+## Enabling Contextual Logs in Error Handling
+
+By default, Angular uses its own `ErrorHandler` to handle errors that occur during server-side rendering. However, this only prints errors to the console with a lack of context, and creates multiline messages that are not parsed correctly by monitoring tools. To improve error handling, Spartacus provides a `CxErrorHandler` that extends the default Angular `ErrorHandler` and uses `LoggerService` to log errors with the relevant context, when available.
+
+To use `CxErrorHandler`, you need to import the `ErrorHandlingModule`. It is recommended that you import the `ErrorHandlingModule` in `spartacus.module.ts`, as shown in the following example:
+
+```ts
+import { ErrorHandlingModule } from '@spartacus/setup/ssr';
+
+@NgModule({
+  imports: [
+    [...]
+    ErrorHandlingModule.forRoot()
+  ]
+})
+export class SpartacusModule {}
+```
+
+By importing `ErrorHandlingModule`, the `CxErrorHandler` then handles all errors that occur during server-side rendering, and these errors are logged with an appropriate context.
+
+**Note:** Since user applications may contain their own implementations of `ErrorHandler`, the `CxErrorHandler` is not used by default when contextual logging is enabled. To use the `CxErrorHandler`, it is recommended that you extend your error handler to use the `LoggerService`.
+
+## LoggerService and Third-party Party Libraries
+
+To ensure that you have context and proper formatting for the logs that are output by third-party libraries in your application, it is recommended that you verify whether custom loggers can be provided, and that you use the `LoggerService` if possible. Otherwise, the logs from third-party libraries will be written in plain text, without the request's context and without proper formatting.
+
+The `LoggerService` can be used not only to log messages that are added to your applications, but also to log messages that are added to those third-party libraries that are used in your project and that communicate by their state.
+
```

Comment:
I tried to adjust the wording of this sentence, but I may have unintentionally changed the meaning! Can you check if it makes sense, or maybe needs rewriting a bit more? Thank you 🙂

Reply:
In my opinion, it makes sense :) thank you!